docs/api.yaml

```diff
@@ -0,0 +1,844 @@
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+    post:
+      tags:
+        - "Student"
+      summary: "Create new students"
+      description: ""
+      operationId: "createStudents"
+      requestBody:
+        description: "Collection of students to create"
+        required: true
+        content:
+          application/json:
+            schema:
+              type: array
+              items:
+                $ref: '#/components/schemas/CreateStudent'
+      responses:
+        200:
+          description: "Collection of created students"
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Student'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+    patch:
+      tags:
+        - "Student"
+      summary: "Update an existing student"
+      parameters:
+        - name: "id"
+          in: "query"
+          description: "Identifier of the student to update"
+          required: true
+          schema:
+            type: string
+      description: |
+        Possible udpates on:
+        * Groups: change
+        * Status: change
+        * EnrolledCourses: add
+      operationId: "updateStudent"
+      requestBody:
+        description: "Student to update"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Student'
+            examples:
+              addGroup:
+                summary: Change group
+                #  description: ""
+                value:
+                  group:
+                    id: "uuid"
+                    name: "Groupe A"
+              changeStatus:
+                summary: Change Status
+                #description: ""
+                value:
+                  status: "ENABLED"
+              addEnrolledCourse:
+                summary: Add Enrolled Course
+                value:
+                  course_id: "uuid"
+                  course_name: "Architecture logicielle"
+      responses:
+        200:
+          description: "The updated Student"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Student'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+  /courses:
+    get:
+      tags:
+        - "Course"
+      summary: "Find courses with optionnal search criteria"
```

Comment:
Find courses with... --> Find courses based on search criteria. Also update other corresponding definition for students and teachers